MSAL/test/unit/MSALPublicClientApplicationTests.m

```diff
@@ -1363,6 +1539,81 @@ - (void)testRemoveAccount_whenAccountExists_shouldRemoveAccount
+
+    NSError *error = nil;
+    BOOL result = [self.tokenCacheAccessor saveTokensWithConfiguration:configuration
+                                                              response:msidResponse
+                                                               factory:[MSIDAADV2Oauth2Factory new]
+                                                               context:nil
+                                                                 error:&error];
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    // 2. Create PublicClientApplication for a different app
+    NSArray *override = @[ @{ @"CFBundleURLSchemes" : @[@"msalmyclient"] } ];
+    [MSALTestBundle overrideObject:override forKey:@"CFBundleURLTypes"];
+
+    MSALPublicClientApplication *application = [[MSALPublicClientApplication alloc] initWithClientId:@"myclient" error:nil];
+    application.tokenCache = self.tokenCacheAccessor;
+    [self.tokenCacheAccessor updateAppMetadataWithFamilyId:@"1" clientId:@"myclient" authority:configuration.authority context:nil error:nil];
+
+    MSIDAuthority *authority = [MSIDAuthorityFactory authorityFromUrl:[NSURL URLWithString:authorityUrl] context:nil error:nil];
+
+    configuration = [[MSIDConfiguration alloc] initWithAuthority:authority
+                                                     redirectUri:UNIT_TEST_DEFAULT_REDIRECT_URI
+                                                        clientId:@"myclient"
+                                                          target:@"fakescope1 fakescope2"];
+
+    MSIDAccount *account = [[MSIDAADV2Oauth2Factory new] accountFromResponse:msidResponse
+                                                               configuration:configuration];
+    MSALAccount *msalAccount = [[MSALAccount alloc] initWithMSIDAccount:account];
+
+    XCTAssertEqualObjects([application allAccounts:nil][0], msalAccount);
+
+    // 3. Remove account
+    result = [application removeAccount:msalAccount error:&error];
+
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    // 4. Make sure the account is now gone
+    XCTAssertEqual([application allAccounts:nil].count, 0);
+
+    // 5. Make sure account and FOCI tokens are still in cache
+    MSIDAccount *cachedAccount = [self.tokenCacheAccessor getAccountForIdentifier:account.accountIdentifier authority:account.authority context:nil error:nil];
+    XCTAssertNotNil(cachedAccount);
+
+    MSIDRefreshToken *fociToken = [self.tokenCacheAccessor getRefreshTokenWithAccount:account.accountIdentifier familyId:@"1" configuration:configuration context:nil error:nil];
+    XCTAssertNotNil(fociToken);
+
+    MSIDRefreshToken *mrrtToken = [self.tokenCacheAccessor getRefreshTokenWithAccount:account.accountIdentifier familyId:nil configuration:configuration context:nil error:nil];
+    XCTAssertNil(mrrtToken);
+
+    [self msalAddDiscoveryResponse:authorityUrl appendDefaultHeaders:YES];
+
+    // 5. Try to acquire token silently
```

Comment:
nit: additional comment on what this indicates? i.e./ ... indicating foci is not being used? or something in lines of that

Reply:
Added a comment